pyciemss/interfaces.py

```diff
@@ -228,10 +228,18 @@ def sample(
 
     Returns:
         result: Dict[str, torch.Tensor]
-            - Dictionary of outputs from the model.
-                - Each key is the name of a parameter or state variable in the model.
-                - Each value is a tensor of shape (num_samples, num_timepoints) for state variables
+            - Dictionary of outputs with following attributes:
+                - data: The samples from the model as a pandas DataFrame.
+                - unprocessed_result: Dictionary of outputs from the model.
+                    - Each key is the name of a parameter or state variable in the model.
+                    - Each value is a tensor of shape (num_samples, num_timepoints) for state variables
                     and (num_samples,) for parameters.
+                - quantiles: The quantiles for ensemble score calculation as a pandas DataFrames.
+                - risk: Dictionary with each key as the name of a state with
+                a dictionary of risk estimates for each state at the final timepoint.
+                    - risk: Estimated alpha-superquantile risk with alpha=0.95
```

Comment:
Is this a hardcoded result, or is it possible to specify different definitions of risk that is returned?

Reply:
also, exactly what is an alpha superquantile, and is there a way to describe it that doesn't require that you already know what an alpha superquantile is?

Reply:
We could add more flexibility to `alpha` value. At the moment, it was decided that we don't need to ask the user for this value so it was hardcoded into the risk estimate.
I am not sure what you mean about describing alpha-superquantile. The users are not really picking a risk measure at the moment so we decided on what to output. However, we are able to estimate many different risk measures. I can add all the statistics that we have available and let the user chose what they want to see.

Reply:
don't get me wrong, I think `alpha=0.95` is a perfectly reasonable default, but I am always leary of hard-coded values.

Here is what chatGPT says about alpha superquantiles:

An alpha superquantile, also known as a conditional value-at-risk (CVaR), expected shortfall, or tail conditional expectation, is a risk measure that goes beyond the traditional value-at-risk (VaR) in financial risk management. It provides a more comprehensive view of the potential risk by considering not only the threshold at which a certain percentage of losses is exceeded (as VaR does) but also the average of the losses that occur beyond that threshold. This makes it particularly useful for evaluating the tail risk of a distribution, where the most extreme losses lie.

Here's a more detailed explanation:

- **Value-at-Risk (VaR)**: VaR is a threshold with a certain confidence level (say, 95% or 99%) under which the potential loss on an investment is expected to fall with that confidence level over a specified period. For example, a 95% VaR of $100 million means that there is a 95% chance that the loss will not exceed $100 million.

- **Alpha Superquantile (Conditional Value-at-Risk, CVaR)**: Given a level \(\alpha\) (e.g., 95% or 99%), the alpha superquantile (CVaR\(_\alpha\)) is the expected loss given that the loss is greater than the VaR\(_\alpha\). In other words, it's the average of the worst \((1-\alpha)\%\) of outcomes. CVaR provides a more accurate and robust sense of risk because it takes into account the severity of losses in the tail of the distribution beyond the VaR threshold.

Mathematically, for a loss random variable \(L\) and a given confidence level \(\alpha\), the CVaR is defined as:
\[ \text{CVaR}_\alpha = \mathbb{E}[L | L > \text{VaR}_\alpha] \]

This measure is widely used in finance, insurance, and other fields where understanding and managing the extreme risks is crucial. It helps in making informed decisions by considering not just how often losses might exceed a certain threshold, but also how severe those losses could be.

I am just saying that alpha superquantile is not an informative name, whereas "average of the worst outcomes" is a bit more descriptive.

Reply:
I added a more rigorous definition of superquantiles and made alpha a user-input.